piet/src/text.rs

```diff
@@ -306,17 +306,24 @@ pub struct LineMetric {
     /// Includes trailing whitespace.
     pub end_offset: usize,
 
-    /// Length in (in UTF-8 code units) of current line's trailing whitespace.
+    /// The length of the trailing whitespace at the end of this line, in utf-8 code units.
     pub trailing_whitespace: usize,
 
-    /// Distance of the baseline from the top of the line
+    /// The distance from the top of the line (`y_offset`) to the baseline.
     pub baseline: f64,
 
-    /// Line height
+    /// The height of the line.
     pub height: f64,
 
     /// Cumulative line height (includes previous line heights)
+    #[deprecated(since = "0.2.0", note = "use y_offset instead")]
```

Comment:
The main reason we went with cumulative height is that all values provide useful information. By measuring the top, the first value is always 0 (I think; there could be handling of padding that makes this not so), and the height can be recovered by subtracting the value for the previous line (unwrap_or 0). But I'm open to the argument that being able to recover height is not important, we don't care too much about the 8 bytes.

Reply:
on macOS, the y_offset of the first line is not 0; I believe that using `y_offset` lets us still recover all values (since y_offest + height == cumulative height) but is more ergonomic in more cases?

Reply:
also: the main reason I like this formulation is that it simplifies finding the baseline, which is now just `y_offset + baseline`, instead of `cumulative_height - height + baseline`.